docs/vocabulary.md

```diff
@@ -388,6 +389,75 @@ pipeline in corresponding documentation.
+
+While CI/CD Tools and Technologies generally give developers broad leeway in naming and implementing their Stages, agreeing upon some common terms will help develop more pluggable Pipelines.
+
+The concepts below can be re-ordered in Pipeline implementations. For example, it is often the case that software linting (logically a test) is done prior to the Build Stage. Also, deployment to a staging environment is often done prior to the Release Stage, while the deployment for production often happens after the Release Stage.
+
+In the lists and table below, Software Artifacts includes: Documentation Source Files, Source Code, Baseline/Composition/Dependency Information, Infrastructure as Code.
+
+"Software Source" refers to human-readable source files that are inputs to the pipeline, such as: Source Code, Configuration Files, Documentation Source Files, Declared Dependencies, Baseline/Composition Information (i.e. lock files in source control)
+
+"Binary Source" refers to executables that are inputs to the pipeline, such as: Executable Software Dependencies, Container Images, Virtual Machine Images
+
+"Generated Software" refers to human-readable files that are outputs of the pipeline, such as: Compiled Software, Generated Configuration Files, Compiled Documentation, Baseline/Composition Information generated by the build. 
+
+"Generated Binaries" refers to executables that are outputs of the pipeline, such as: Executable Software, Container Images, Virtual Machine Images
+
+#### Build Stage
+* Semantics: Set up the build workspace and provision build resources. Download, retrieve, assemble, and/or compile software into an executable and testable format. Download, retrieve, assemble, and/or compile documentation into a consumable format.
+* Aliases: Compile
+* Inputs: Software Source, Binary Source
+* Outputs: Generated Software, Generated Binaries
+* Other Results and Side Effects: The build may also capture point-in-time details about the build environment and tools (compiler versions, package manager versions, task container versions in tekton, OS versions etc.) for anyone who wants to rebuild or understand a build.
+
+#### Test Stage
+* Semantics: Test, scan, verify, and lint software and documentation.
+* Aliases: Verify
+* Inputs: Software Source, Binary Source, Generated Software, Generated Binaries
+* Outputs: Test/Lint/Scan Records, Test/Lint/Scan Reports, Test/Lint/Scan Coverage Reports
+* Other Results and Side Effects: N/A
+
+#### Release Stage
+* Semantics: Package, version, sign, and publish software artifacts and documentation.
+* Aliases: Deliver, Publish
+* Inputs: Software Source, Binary Source, Generated Software, Generated Binaries
+* Outputs: Release Records, Release Reports
+* Other Results and Side Effects: Documentation and Software: packaged, signed, and published to a repository.
```

Comment:
I guess this stage would also produce Generated software and Generated Binaries? E.g. a tar.gz file or a BOM (bill of materials) file

Reply:
Yep.  Added.